ibis-server/app/routers/v2/connector.py

```diff
@@ -59,10 +69,43 @@ async def query(
             java_engine_connector=java_engine_connector,
         ).rewrite(sql)
         connector = Connector(data_source, dto.connection_info)
+
+        # First check if the query is a dry run
+        # If it is dry run.
+        # We don't need to check query cache
         if dry_run:
             connector.dry_run(rewritten_sql)
-            return Response(status_code=204)
-        return ORJSONResponse(to_json(connector.query(rewritten_sql, limit=limit)))
+            dry_response = Response(status_code=204)
+            dry_response.headers["X-Cache-Hit"] = "true"
```

Comment:
It's not right. Dry run success doesn't mean the cache hit. It only validates the SQL; we shouldn't return the cache hit header.